client/pages/offer-thread.jsx

```diff
@@ -1,42 +1,66 @@
 
 import React from 'react';
 import io from 'socket.io-client';
+let buyerId = null;
+let sellerId = null;
 
 class OfferThread extends React.Component {
+
   constructor(props) {
     super(props);
     this.state = {
       messageIn: null,
-      reply: ''
+      user: null,
+      post: null,
+      reply: '',
+      socketThread: null
     };
     this.handleSubmit = this.handleSubmit.bind(this);
     this.handleMessage = this.handleMessage.bind(this);
+    this.scrollToBottom = this.scrollToBottom.bind(this);
   }
 
   componentDidMount() {
+    this.scrollToBottom();
     const token = window.localStorage.getItem('user-jwt');
     fetch(`/api/messages/${this.props.postId}/${this.props.senderId}`, {
       headers: {
         'X-Access-Token': token
       }
-    })
+    }
+
+    )
       .then(response => response.json())
       .then(offerThread => {
-        this.setState({ messageIn: offerThread });
+        this.setState({
+          messageIn: offerThread.messages,
+          user: offerThread.user,
+          post: offerThread.post
+        });
+        buyerId = offerThread.user.userId;
+        sellerId = offerThread.post.userId;
+        const socket = io.connect('/', {
+          transports: ['websocket'],
```

Comment:
Store the `socket` on `this` so you can disconnect it when the user leaves this screen.

```js
this.socket = io.connect('/', {
```